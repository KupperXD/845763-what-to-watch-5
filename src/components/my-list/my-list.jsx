import React from 'react';
import {Link} from 'react-router-dom';
import PropType from 'prop-types';
import {filmType} from '../../types/index';


const MyList = (props) => {
  const {name, picture, id} = props.film; 


  return (
    <React.Fragment>
      <div className="user-page">
        <header className="page-header user-page__head">
          <div className="logo">
            <a href="main.html" className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <h1 className="page-title user-page__title">My list</h1>

          <div className="user-block">
            <div className="user-block__avatar">
              <img src="/img/avatar.jpg" alt="User avatar" width="63" height="63"/>
            </div>
          </div>
        </header>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <div className="catalog__movies-list">
            <article className="small-movie-card catalog__movies-card">
              <div className="small-movie-card__image">
                <img src={picture}
                  alt={name} width="280" height="175"/>
              </div>
              <h3 className="small-movie-card__title">
                <Link to={`/films/${id}`} className="small-movie-card__link">{name}</Link>
              </h3>
            </article>
          </div>
        </section>

        <footer className="page-footer">
          <div className="logo">
            <a href="main.html" className="logo__link logo__link--light">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <div className="copyright">
            <p>Â© 2019 What to watch Ltd.</p>
          </div>
        </footer>
      </div>
    </React.Fragment>);
};

MyList.propType = {
  film: PropType.shape(filmType).isRequired,
}

export default MyList;
