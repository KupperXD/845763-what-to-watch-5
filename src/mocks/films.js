export default [
  {
    backgroundColor: `#73B39A`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    id: 1,
    isFavorite: true,
    name: `Bronson`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 3.6,
    released: 2008,
    runTime: 92,
    scoresCount: 109661,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`
  }
];
