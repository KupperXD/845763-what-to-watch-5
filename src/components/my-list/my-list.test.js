import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";
import films from "../../mocks/films";

import {MyList} from "./my-list";

const noop = () => {};


it(`renders correctly`, () => {
  const myList = renderer
    .create(
        <BrowserRouter>
          <MyList
            user={{}}
            favorites={films}
            loadFavorites={noop}
          />
        </BrowserRouter>
    )
    .toJSON();

  expect(myList).toMatchSnapshot();
});
