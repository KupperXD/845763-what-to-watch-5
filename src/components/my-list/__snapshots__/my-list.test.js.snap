// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--undefined"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/films/1"
          onClick={[Function]}
        >
          <video
            autoPlay={false}
            controls={false}
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
            style={
              Object {
                "maxWidth": "100%",
                "objectFit": "cover",
              }
            }
          >
            <source
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              type="video/webm"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            Bronson
          </h3>
        </a>
      </article>
    </div>
    <div
      className="catalog__more"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <button
        className="catalog__button"
        onClick={[Function]}
        type="button"
      >
        Show more
      </button>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
